src/Module.zig

```diff
@@ -836,6 +836,31 @@ pub const ErrorSet = struct {
             .lazy = .{ .node_offset = self.node_offset },
         };
     }
+
+    /// sort the NameMap. This should be called whenever the map is modified.
+    /// alloc should be the allocator used for the NameMap data.
+    pub fn sortNames(names: *NameMap, alloc: Allocator) !void {
+        // Perform an insertion sort over the entries based on key.
+        // We ignore value cause its always void.
+        const list = &names.entries;
+        const keys = list.items(.key);
+        const hashes = list.items(.hash);
+
+        var i: usize = 1;
+        while (i < keys.len) : (i += 1) {
+            const key = keys[i];
+            const hash = hashes[i];
+            var j: usize = i;
+            while (j > 0 and std.mem.lessThan(u8, key, keys[j - 1])) : (j -= 1) {
+                keys[j] = keys[j - 1];
+                hashes[j] = hashes[j - 1];
+            }
+            keys[j] = key;
+            hashes[j] = hash;
+        }
+
+        try names.reIndex(alloc);
+    }
```

Comment:
I made #11109 so that this can be improved:

```suggestion
    pub fn sortNames(names: *NameMap) void {
        const Context = struct {
            keys: [][]const u8,

            pub fn lessThan(ctx: @This(), a_index: usize, b_index: usize) bool {
                return std.mem.lessThan(u8, ctx.keys[a_index], ctx.keys[b_index]);
            }
        };
        names.sort(Context{ .keys = names.keys() });
    }
```

Reply:
Looks great! Once #11109 merges I'll make those changes.

Reply:
Done!